pkg/pgmodel/end_to_end_tests/promql_endpoint_integration_test.go

```diff
@@ -11,23 +13,82 @@ import (
 	"github.com/prometheus/common/model"
 )
 
-type sample struct {
-	Metric model.Metric       `json:"metric"`
-	Values []model.SamplePair `json:"values"`
-	Value  model.SamplePair   `json:"value"`
-}
-
 type requestCase struct {
 	req *http.Request
 	log string
 }
 
+type sample struct {
+	Timestamp int64
+	Raw       string
+	Metric    model.Metric       `json:"metric"`
+	Values    []model.SamplePair `json:"values"`
+}
+
+type dummySample sample
+
+func (s *sample) UnmarshalJSON(data []byte) error {
+	var temp dummySample
+	if err := json.Unmarshal(data, &temp); err != nil {
+		if err = json.Unmarshal(data, &s.Timestamp); err != nil {
+			return json.Unmarshal(data, &s.Raw)
+		}
+		return nil
+	}
+	*s = sample(temp)
+	return nil
```

Comment:
Is this really the best way to write this code?
What do you think of something like
```suggestion
	var temp dummySample
	err := json.Unmarshal(data, &temp)
	if == nil {
		*s = sample(temp)
		return nil
	}
	err = json.Unmarshal(data, &s.Timestamp)
	if err == nil {
		return nil
	}
	return json.Unmarshal(data, &s.Raw)
```
?

Reply:
Yup, updated.